tendermint/src/validator.rs

```diff
@@ -51,14 +49,25 @@ impl From<Set> for RawValidatorSet {
 }
 
 impl Set {
-    /// constructor
+    /// Constructor
     pub fn new(
         validators: Vec<Info>,
         proposer: Option<Info>,
-        total_voting_power: vote::Power,
+        total_voting_power: Option<vote::Power>,
```

Comment:
why would someone need to pass `total_voting_power` if it can be computed?

Reply:
Yeah I wonder about this as well. Maybe @ebuchman knows why it was done like this?

Reply:
they do not I've been campaigning against total voting power and proposer fields in this type for a few weeks now! I suspect it was done without really thinking about it ...